### PR TITLE
Raise error instead of writing to log

### PR DESCRIPTION
**Pull request by ggravlingen**
Coverage dropped because I've added the pytradfri directory rather than only testing the tests-folder.

### CODE REVIEW

File: tests/test_light.py [hunk 1]
```diff
@@ -233,7 +233,3 @@ def test_setters():
     cmd = Device(LIGHT_CWS).light_control \
         .set_predefined_color('Warm glow')
     assert cmd.data == {'3311': [{'5706': 'efd275'}]}
-
-    cmd = Device(LIGHT_CWS).light_control \
```
Comment (balloob) [inline tests/test_light.py]: Add a test that tests that it raises:

```
import pytest
from pytradfri import error

with pytest.raises(error.InvalidColor):
    Device(…).light…
```

File: pytradfri/device.py [hunk 2]
```diff
@@ -221,7 +220,7 @@ def set_predefined_color(self, colorname, *, index=0):
             color = COLORS[colorname.lower().replace(" ", "_")]
             return self.set_hex_color(color, index=index)
         except KeyError:
-            _LOGGER.debug('Could not match color name')
+            raise PyTradFriError("Could not match color name in COLOR_NAMES")
```
Comment (balloob) [inline pytradfri/device.py]: Either create a new exception that inherits from this general error (ie `InvalidColor` error) or use the built-in exception for invalid values: `ValueError`.

File: pytradfri/device.py [hunk 3]
```diff
@@ -221,7 +220,8 @@ def set_predefined_color(self, colorname, *, index=0):
             color = COLORS[colorname.lower().replace(" ", "_")]
             return self.set_hex_color(color, index=index)
         except KeyError:
-            _LOGGER.debug('Could not match color name')
+            raise ColorError('Unable to match %s in COLOR_NAMES',
```
Comment (balloob) [inline pytradfri/device.py]: Error message leaks implementation detail. I would go for "Invalid color specified: %s"

**Comment by ggravlingen** [comment]
@balloob Thanks! I'll squash and merge as soon as Travis starts working again and tests pass.

**Pull request merged**